python/paddle/fluid/compiler.py

```diff
@@ -122,6 +134,11 @@ def __init__(self, program_or_graph):
         self._compiled = False
         self._is_data_parallel = False
         self._is_inference = False
+        self._loss_name = None
+        self._share_vars_from = None
+        self._places = None
+        self._build_strategy = build_strategy
+        self._exec_strategy = None
 
     def with_data_parallel(self,
```

Comment:
I think we can set the `is_data_parallel`, `is_inference` to the arguments of `CompiledProgram`, now if user just use `CompiledProgram` but not invoke `with_data_parallel`, the program is still executed by `Executor`, In this case, the program is not `compiled` in fact.

Reply:
> I think we can set the is_data_parallel, is_inference to the arguments of CompiledProgram

I don't think this is better.

> now if user just use CompiledProgram but not invoke with_data_parallel, the program is still executed by Executor, In this case, the program is not compiled in fact.

Please read this introduction: https://github.com/PaddlePaddle/Paddle/pull/18919#issue-302769160